src/Nancy.Hosting.Self/FileSystemRootPathProvider.cs

```diff
@@ -1,17 +1,34 @@
 namespace Nancy.Hosting.Self
 {
-    using System.IO;
-    using System.Reflection;
+#if DNX451
+    using Microsoft.Framework.Runtime;
+    using Microsoft.Framework.Runtime.Infrastructure;
+#endif
 
     public class FileSystemRootPathProvider : IRootPathProvider
     {
+        private readonly IApplicationEnvironment env;
+
+        public FileSystemRootPathProvider()
+        {
+#if DNX451
+            this.env = CallContextServiceLocator.Locator.ServiceProvider.GetService(typeof(IApplicationEnvironment)) as IApplicationEnvironment;

```

Comment:
I'd rather see this factored out into a separate service so it could be mocked and tested.